diff --git a/st-13.d.ts b/st-13.d.ts
index 73d5dc41635b85a05f7840f1d321a168b036f02c..eafb5e9e29d97da98514dd3ec6aa049bb90748af 100644
--- a/st-13.d.ts
+++ b/st-13.d.ts
@@ -1492,4 +1492,11 @@ interface BoxLayout extends Atk.ImplementorIface, Clutter.Animatable, Clutter.Co
     connect_after(sigName: string, callback: (...args: any[]) => void): number
     emit(sigName: string, ...args: any[]): void
     disconnect(id: number): void
+
+    ease(props: {
+        readonly x?: number
+        readonly y?: number
+        readonly duration?: number
+        readonly mode?: Clutter.AnimationMode
+    }): void
 }
